magento1-module/app/code/local/Divante/VueStorefrontBridge/controllers/RewriteController.php

```diff
@@ -0,0 +1,27 @@
+<?php
+require_once Mage::getModuleDir('controllers', 'Divante_VueStorefrontBridge') . DS . 'AbstractController.php';
+class Divante_VueStorefrontBridge_RewriteController extends Divante_VueStorefrontBridge_AbstractController {
+
+    public function targetAction()
+    {
+
+        if (!$this->_checkHttpMethod('GET')) {
+            return $this->_result(500, 'Only GET method allowed');
+        }
+        $requestPath = $this->getRequest()->getParam('request_path');
+        if ($requestPath) {
+
+            $reader = Mage::getSingleton('core/resource')->getConnection('core_read');
+            $select = $reader->select()
+                             ->from('core_url_rewrite', ['target_path'])->where('request_path = ?', $requestPath);
```

Comment:
Please add a `LIMIT 1` to the query to improve performance.

Reply:
I honestly thought that ->fetchOne() would do exactly like that but you get surprised by magento/Zend/Varien internals every now and then:) Limiting beforehand now introduced.

Reply:
I know, it really sucks :/
Thanks!